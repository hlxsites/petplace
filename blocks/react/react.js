export default async function decorate(block) {
  block.id = 'react-root';
  block.innerHTML = '';

  import('./react-index-Gmzjj7aU.js');
}
