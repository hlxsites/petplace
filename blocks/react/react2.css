.uppy-Root{position:relative;box-sizing:border-box;color:#333;font-family:-apple-system,system-ui,BlinkMacSystemFont,Segoe UI,Segoe UI Symbol,"Segoe UI Emoji","Apple Color Emoji",Roboto,Helvetica,Arial,sans-serif;line-height:1;text-align:left;-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale}[dir=rtl] .uppy-Root,.uppy-Root[dir=rtl]{text-align:right}.uppy-Root *,.uppy-Root *:before,.uppy-Root *:after{box-sizing:inherit}.uppy-Root [hidden]{display:none}.uppy-u-reset{all:initial;box-sizing:border-box;-webkit-appearance:none;-moz-appearance:none;appearance:none;font-family:-apple-system,system-ui,BlinkMacSystemFont,Segoe UI,Segoe UI Symbol,"Segoe UI Emoji","Apple Color Emoji",Roboto,Helvetica,Arial,sans-serif;line-height:1}[dir=rtl] .uppy-u-reset{text-align:right}.uppy-c-textInput{font-family:inherit;font-size:14px;line-height:1.5;padding:6px 8px;background-color:#fff;border:1px solid #ddd;border-radius:4px}.uppy-size--md .uppy-c-textInput{padding:8px 10px}.uppy-c-textInput:focus{border-color:#1269cf99;outline:none;box-shadow:0 0 0 3px #1269cf26}[data-uppy-theme=dark] .uppy-c-textInput{color:#eaeaea;background-color:#333;border-color:#333}[data-uppy-theme=dark] .uppy-c-textInput:focus{border-color:#525252;box-shadow:none}.uppy-c-icon{display:inline-block;max-width:100%;max-height:100%;overflow:hidden;fill:currentColor}.uppy-c-btn{display:inline-flex;justify-content:center;align-items:center;font-weight:500;font-size:inherit;font-family:inherit;color:inherit;line-height:1;white-space:nowrap;text-align:center;transition-duration:.3s;transition-property:background-color,color;-webkit-user-select:none;-moz-user-select:none;user-select:none}[dir=rtl] .uppy-c-btn{text-align:center}.uppy-c-btn:not(:disabled):not(.disabled){cursor:pointer}.uppy-c-btn::-moz-focus-inner{border:0}.uppy-c-btn-primary{padding:10px 18px;color:#fff;font-size:14px;background-color:#1269cf;border-radius:4px}.uppy-c-btn-primary:not(:disabled):hover{background-color:#0e51a0}.uppy-c-btn-primary:focus{outline:none;box-shadow:0 0 0 3px #1269cf66}.uppy-size--md .uppy-c-btn-primary{padding:13px 22px}[data-uppy-theme=dark] .uppy-c-btn-primary{color:#eaeaea}[data-uppy-theme=dark] .uppy-c-btn-primary:focus{outline:none}[data-uppy-theme=dark] .uppy-c-btn-primary::-moz-focus-inner{border:0}[data-uppy-theme=dark] .uppy-c-btn-primary:focus{box-shadow:0 0 0 2px #aae1ffd9}.uppy-c-btn-primary.uppy-c-btn--disabled{background-color:#8eb2db}.uppy-c-btn-link{padding:10px 15px;color:#525252;font-size:14px;line-height:1;background-color:transparent;border-radius:4px}.uppy-c-btn-link:hover{color:#333}.uppy-c-btn-link:focus{outline:none;box-shadow:0 0 0 3px #1269cf40}.uppy-size--md .uppy-c-btn-link{padding:13px 18px}[data-uppy-theme=dark] .uppy-c-btn-link{color:#eaeaea}[data-uppy-theme=dark] .uppy-c-btn-link:focus{outline:none}[data-uppy-theme=dark] .uppy-c-btn-link::-moz-focus-inner{border:0}[data-uppy-theme=dark] .uppy-c-btn-link:focus{box-shadow:0 0 0 2px #aae1ffd9}[data-uppy-theme=dark] .uppy-c-btn-link:hover{color:#939393}.uppy-DragDrop-container{display:flex;align-items:center;justify-content:center;max-width:100%;font-family:-apple-system,system-ui,BlinkMacSystemFont,Segoe UI,Segoe UI Symbol,"Segoe UI Emoji","Apple Color Emoji",Roboto,Helvetica,Arial,sans-serif;background-color:#fff;border-radius:7px;cursor:pointer}.uppy-DragDrop-container::-moz-focus-inner{border:0}.uppy-DragDrop-container:focus{outline:none;box-shadow:0 0 0 3px #1269cf66}.uppy-DragDrop-inner{margin:0;padding:80px 20px;line-height:1.4;text-align:center}.uppy-DragDrop-arrow{width:60px;height:60px;margin-bottom:17px;fill:#e0e0e0}.uppy-DragDrop--isDragDropSupported{border:2px dashed #adadad}.uppy-DragDrop--isDraggingOver{background:#eaeaea;border:2px dashed #1269cf}.uppy-DragDrop--isDraggingOver .uppy-DragDrop-arrow{fill:#939393}.uppy-DragDrop-label{display:block;margin-bottom:5px;font-size:1.15em}.uppy-DragDrop-browse{color:#1269cf;cursor:pointer}.uppy-DragDrop-note{color:#adadad;font-size:1em}
