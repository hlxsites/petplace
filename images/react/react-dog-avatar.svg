<svg width="276" height="276" viewBox="0 0 276 276" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M0 138C0 61.7847 61.7847 0 138 0C214.215 0 276 61.7847 276 138C276 214.215 214.215 276 138 276C61.7847 276 0 214.215 0 138Z" fill="#EBE8FC"/>
<g clip-path="url(#clip0_8102_1440)">
<path fill-rule="evenodd" clip-rule="evenodd" d="M138.627 93.0727H157.775C157.882 92.8656 157.997 92.6461 158.121 92.4153C159.001 90.7769 160.331 88.5496 162.149 86.1615C165.585 81.6462 170.924 76.3494 178.445 73.5672C178.989 73.2092 179.64 73.0009 180.34 73.0009H180.576C180.628 72.9997 180.681 72.9997 180.733 73.0009H208.566C210.471 73.0009 212.016 74.5455 212.016 76.4508V104.677C212.016 106.582 210.471 108.127 208.566 108.127C207.744 108.127 206.936 108.097 206.143 108.039L195.35 136.686L195.35 136.686C194.964 137.712 194.766 138.794 194.766 139.888V168.971C194.766 184.208 179.781 194.374 163.09 194.374C162.131 194.374 161.19 194.321 160.268 194.221V196.79C160.268 209.084 150.301 219.05 138.008 219.05C125.714 219.05 115.747 209.084 115.747 196.79V194.22C114.821 194.321 113.875 194.374 112.91 194.374C96.2193 194.374 81.2337 184.208 81.2337 168.971V139.814C81.2337 138.761 81.0509 137.72 80.6928 136.729L69.8591 108.039C69.0655 108.097 68.2572 108.127 67.4343 108.127C65.5289 108.127 63.9844 106.582 63.9844 104.677V76.4508C63.9844 74.5455 65.5289 73.0009 67.4343 73.0009H95.6606C95.767 73.0009 95.8723 73.0057 95.9762 73.0152C96.4295 72.9737 96.8984 73.0218 97.3586 73.1718C105.483 75.8212 111.186 81.4223 114.794 86.1624C116.612 88.5505 117.942 90.7778 118.822 92.4162C118.946 92.647 119.061 92.8665 119.168 93.0736H138.627V93.0727ZM144.934 187.794C147.364 189.635 150.196 191.332 153.368 192.541V196.79C153.368 205.274 146.491 212.151 138.008 212.151C129.524 212.151 122.647 205.274 122.647 196.79V192.535C125.813 191.326 128.64 189.632 131.067 187.794C132.321 186.844 133.485 185.843 134.551 184.829L134.551 190.412C134.551 192.317 136.095 193.862 138.001 193.862C139.906 193.862 141.45 192.317 141.45 190.412L141.45 184.83C142.516 185.843 143.68 186.844 144.934 187.794ZM76.7463 106.746L87.1557 134.312L87.1703 134.352C87.8072 136.103 88.1334 137.949 88.1334 139.814V168.971C88.1334 178.824 98.2379 187.474 112.91 187.474C118.201 187.474 122.975 185.268 126.9 182.294C130.412 179.634 133.056 176.497 134.55 174.245V163.681C130.957 163.063 128.526 161.467 127.008 159.443C126.087 158.215 125.624 156.985 125.392 156.053C125.275 155.586 125.213 155.182 125.18 154.874C125.163 154.719 125.153 154.586 125.148 154.479C125.145 154.426 125.144 154.378 125.143 154.338L125.142 154.282L125.142 154.258L125.142 154.247L125.142 154.242C125.142 154.239 125.142 154.236 128.591 154.236H125.142C125.142 152.331 126.686 150.787 128.591 150.787H147.409C149.314 150.787 150.859 152.331 150.859 154.236H147.409C150.859 154.236 150.859 154.239 150.859 154.242L150.859 154.247L150.859 154.258L150.859 154.282L150.858 154.338C150.857 154.378 150.855 154.426 150.852 154.479C150.847 154.586 150.837 154.719 150.821 154.874C150.788 155.182 150.726 155.586 150.609 156.053C150.376 156.985 149.914 158.215 148.993 159.443C147.475 161.467 145.043 163.063 141.45 163.681V174.245C142.944 176.497 145.588 179.634 149.1 182.294C153.025 185.268 157.799 187.474 163.09 187.474C177.762 187.474 187.867 178.824 187.867 168.971V139.888C187.867 137.96 188.215 136.055 188.893 134.254L188.894 134.253L199.257 106.747C193.399 104.93 188.614 101.471 185.015 97.2813C181.025 92.6378 178.43 87.0324 177.397 81.6248C173.24 83.9349 170.008 87.2275 167.639 90.3402C166.085 92.382 164.946 94.2906 164.199 95.6806C163.93 96.1821 163.561 97.0233 163.312 97.5922C163.216 97.8096 163.138 97.9873 163.09 98.091H138.627L114.165 98.091C114.076 97.9006 113.807 97.4612 113.515 96.9841L113.515 96.9837L113.515 96.9836L113.515 96.9834L113.515 96.9834C113.232 96.5207 112.927 96.0226 112.744 95.6815C111.997 94.2915 110.857 92.3829 109.304 90.3411C106.774 87.0171 103.26 83.488 98.6869 81.1693C97.7148 86.7165 95.0873 92.5073 90.9856 97.2814C87.3866 101.47 82.6028 104.929 76.7463 106.746ZM184.096 79.9007C184.834 84.194 186.898 88.8851 190.248 92.785C193.831 96.9555 198.808 100.139 205.116 100.997V79.9007H184.096ZM70.8842 79.9007V100.997C77.1922 100.139 82.1689 96.9555 85.7521 92.785C89.1026 88.8851 91.1664 84.194 91.904 79.9007H70.8842ZM164.972 143.253C168.09 143.253 170.617 140.726 170.617 137.608C170.617 134.49 168.09 131.963 164.972 131.963C161.854 131.963 159.327 134.49 159.327 137.608C159.327 140.726 161.854 143.253 164.972 143.253ZM116.673 137.608C116.673 140.726 114.146 143.253 111.028 143.253C107.91 143.253 105.383 140.726 105.383 137.608C105.383 134.49 107.91 131.963 111.028 131.963C114.146 131.963 116.673 134.49 116.673 137.608Z" fill="#302280"/>
</g>
<defs>
<clipPath id="clip0_8102_1440">
<rect width="148" height="148" fill="white" transform="translate(64 73)"/>
</clipPath>
</defs>
</svg>
