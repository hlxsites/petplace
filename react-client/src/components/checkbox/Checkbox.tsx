import * as RadixCheckbox from "@radix-ui/react-checkbox";
import { classNames } from "../../util/styleUtil";
import { Icon } from "../icon/Icon";

export type CheckboxProps = {
  asChild?: boolean;
  checked?: boolean | "indeterminate";
  className?: string;
  defaultChecked?: boolean | "indeterminate";
  disabled?: boolean;
  id: string;
  name: string;
  onCheckedChange: () => void;
  required?: boolean;
  value: string;
};

const Checkbox = ({ className, ...props }: CheckboxProps) => {
  return (
    <RadixCheckbox.Root
      className={classNames(
        "p-0 flex h-5 w-5 items-center justify-center rounded-md border-solid border-[#6E6D73] bg-white focus:outline-none",
        className
      )}
      {...props}
    >
      <RadixCheckbox.Indicator className="absolute flex h-5 w-5 items-center justify-center rounded-md bg-[#C74D3F] text-white">
        <Icon display="check" />
      </RadixCheckbox.Indicator>
    </RadixCheckbox.Root>
  );
};

export default Checkbox;
