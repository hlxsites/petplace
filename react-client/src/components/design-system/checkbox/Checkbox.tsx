import * as RadixCheckbox from "@radix-ui/react-checkbox";
import { ComponentPropsWithoutRef, ElementRef, forwardRef } from "react";

import { classNames } from "~/util/styleUtil";
import {
  ElementInputCheckboxGroup,
  InputWithoutFormBuilderProps,
} from "../form/types/formTypes";
import { Icon } from "../icon/Icon";

type CheckboxVariant = "orange" | "purple";

type CheckboxGroupType =
  InputWithoutFormBuilderProps<ElementInputCheckboxGroup>;

type CheckboxProps = Omit<
  Omit<ComponentPropsWithoutRef<typeof RadixCheckbox.Root>, "onChange">,
  "children"
> &
  Omit<CheckboxGroupType, "value" | "options" | "onChange"> & {
    variant?: CheckboxVariant;
  };

export const Checkbox = forwardRef<
  ElementRef<typeof RadixCheckbox.Root>,
  CheckboxProps
>(({ className, hideLabel, label, variant = "orange", ...props }, ref) => {
  const { indicatorClassName, rootClassName } = useCheckboxVariant(variant);
  const labelElement = (() => {
    if (hideLabel) return null;
    return (
      <label className="text-sm font-franklin" htmlFor={props.id}>
        {label}
      </label>
    );
  })();

  return (
    <div className="grid grid-cols-[auto,_1fr] items-center gap-small">
      <RadixCheckbox.Root
        aria-label={hideLabel ? label : undefined}
        className={classNames(rootClassName, className)}
        ref={ref}
        {...props}
      >
        <RadixCheckbox.Indicator className={indicatorClassName}>
          <Icon display="checkSolo" />
        </RadixCheckbox.Indicator>
      </RadixCheckbox.Root>
      {labelElement}
    </div>
  );
});

function useCheckboxVariant(variant: CheckboxVariant) {
  const rootClassName = classNames(
    "flex h-5 w-5 items-center justify-center rounded-md border-solid border-[#6E6D73] bg-white p-0 focus:outline-none",
    { "focus:bg-purple-300 hover:bg-purple-300": variant === "purple" }
  );

  const indicatorClassName = classNames(
    "absolute flex h-5 w-5 items-center justify-center rounded-md text-white",
    {
      "bg-orange-300-contrast": variant === "orange",
      "bg-purple-300": variant === "purple",
    }
  );

  return { indicatorClassName, rootClassName };
}
