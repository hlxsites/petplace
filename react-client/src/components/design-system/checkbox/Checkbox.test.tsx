import { render, screen } from "@testing-library/react";
import { ComponentProps } from "react";
import userEvent from "@testing-library/user-event";

import { Checkbox } from "./Checkbox";

const { queryByText, getByRole } = screen;

describe("<Checkbox />", () => {
  it(`should render checkbox`, () => {
    getRenderer();
    expect(getByRole("checkbox")).toBeInTheDocument();
  });

  it(`should hide checkbox label when hideLabel is true`, () => {
    getRenderer({ hideLabel: true });
    expect(queryByText("Test label")).not.toBeInTheDocument();
  });

  it.each([false, undefined])(
    `should render checkbox label when hideLabel is %s`,
    (expected) => {
      getRenderer({ hideLabel: expected });
      expect(queryByText("Test label")).toBeInTheDocument();
    }
  );

  it.each(["a-class", "another-class"])(
    `should render custom class '%s'`,
    (expected) => {
      getRenderer({ className: expected });
      expect(getByRole("checkbox")).toHaveClass(expected);
    }
  );

  it(`should render checkbox disabled when disabled is true`, () => {
    getRenderer({ disabled: true });
    expect(getByRole("checkbox")).toBeDisabled();
  });

  it(`should render checkbox unchecked by default`, () => {
    getRenderer();
    const checkbox = getByRole("checkbox");
    expect(checkbox).not.toBeChecked();
  });

  it(`should call event callbacks`, async () => {
    const checkHandler = jest.fn();
    getRenderer({ onCheckedChange: checkHandler });
    const checkbox = getByRole("checkbox");
    expect(checkbox).not.toBeChecked();

    await userEvent.click(checkbox);
    expect(checkbox).toBeChecked();

    await userEvent.click(checkbox);
    expect(checkbox).not.toBeChecked();
    expect(checkHandler).toHaveBeenCalledTimes(2);
  });

  it("should match snapshot to assure that the component is being rendered correctly", () => {
    const { container } = getRenderer();

    expect(container).toMatchSnapshot();
  });
});

// Helpers
type Props = ComponentProps<typeof Checkbox>;
function getRenderer({
  label = "Test label",
  id = "Test id",
  ...rest
}: Partial<Props> = {}) {
  return render(<Checkbox label={label} id={id} {...rest} />);
}
