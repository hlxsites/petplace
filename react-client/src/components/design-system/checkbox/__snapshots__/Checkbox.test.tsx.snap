// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should match snapshot to assure that the component is being rendered correctly 1`] = `
<div>
  <div
    class="flex items-center gap-2"
  >
    <button
      aria-checked="false"
      class="p-0 flex h-5 w-5 items-center justify-center rounded-md border-solid border-[#6E6D73] bg-white focus:outline-none"
      data-state="unchecked"
      id="Test id"
      role="checkbox"
      type="button"
      value="on"
    />
    <label
      class="Label"
      for="Test id"
    >
      Test label
    </label>
  </div>
</div>
`;
